File: Examples/ObjC/VizuryObjCSample/VizuryObjCSample/ViewController.m
//
//  ViewController.m
//  VizuryObjCSample
//
//  Created by Chowdhury Md Rajib  Sarwar on 27/4/20.
//  Copyright © 2020 Chowdhury Md Rajib  Sarwar. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}

- (IBAction)normalEvent:(id)sender {
    NSDictionary *attributeDictionary  =   [[NSDictionary alloc] initWithObjectsAndKeys:
                                            @"AKSJDASNBD",@"productid",
                                            @"789", @"productPrice",
                                            @"Shirt",@"category",
                                            nil];
    
    [VizuryEventLogger logEvent:@"productPage" WithAttributes:attributeDictionary];
}

// you can send the attributes as JSONArray or JSONObject also
- (IBAction)JSONEvent:(UIButton *)sender {
    
    // sending a JSONObject
    NSDictionary *productDetail = [NSDictionary dictionaryWithObjectsAndKeys:
                                   @"62112",@"product_id",
                                   @"1",@"quantity",
                                   @"50", @"price",
                                   nil];
    
    NSData *productjsonData = [NSJSONSerialization dataWithJSONObject:productDetail options:0 error:nil];
    NSString *productjsonStr = [[NSString alloc] initWithData:productjsonData encoding:NSUTF8StringEncoding];
    
    NSDictionary *productjsonDictionary = [NSDictionary dictionaryWithObjectsAndKeys:
                                           productjsonStr,@"viz_data",
                                           nil];
    [VizuryEventLogger logEvent:@"productDetails" WithAttributes:productjsonDictionary];
    
    // sending a JSON Array of Objects
    NSMutableArray *products = [[NSMutableArray alloc] init];
    NSDictionary *productDetail1 = [NSDictionary dictionaryWithObjectsAndKeys:
                                    @"62112",@"product_id",
                                    @"1",@"quantity",
                                    @"50", @"price",
                                    nil];
    
    NSDictionary *productDetail2 = [NSDictionary dictionaryWithObjectsAndKeys:
                                    @"123123",@"product_id",
                                    @"1",@"quantity",
                                    @"50", @"price",
                                    nil];
    
    [products addObject:productDetail1];
    [products addObject:productDetail2];
    
    NSData *wishListjsonData = [NSJSONSerialization dataWithJSONObject:products options:0 error:nil];
    NSString *wishListjsonStr = [[NSString alloc] initWithData:wishListjsonData encoding:NSUTF8StringEncoding];
    
    NSDictionary *wishListjsonDictionary = [NSDictionary dictionaryWithObjectsAndKeys:
                                            wishListjsonStr,@"viz_data",
                                            nil];
    
    [VizuryEventLogger logEvent:@"wishlist" WithAttributes:wishListjsonDictionary];
}

@end

File: Examples/Swift/VizurySwiftSample/VizurySwiftSample/ViewController.swift
//
//  ViewController.swift
//  VizurySwiftSample
//
//  Created by Chowdhury Md Rajib  Sarwar on 27/4/20.
//  Copyright © 2020 Chowdhury Md Rajib  Sarwar. All rights reserved.
//

import UIKit
import VizuryEventLogger

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()

    }

    @IBAction func normalEvent(_ sender: Any) {
        let attributeDictionary = NSDictionary(dictionary: ["productid" : "AKSJDASNBD",
                                                            "productPrice": "789",
                                                            "category": "Shirt"])
        VizuryEventLogger.logEvent("productPage", withAttributes: attributeDictionary as? [AnyHashable : Any])
    }
    
    @IBAction func jsonEvent(_ sender: Any) {
        let productDetail = NSDictionary(dictionary: ["product_id" : "62112",
                                                      "quantity": "1",
                                                      "price": "50"])
        
        var productjsonData = Data()
        do {
        productjsonData = try JSONSerialization.data(withJSONObject: productDetail, options: [])
        } catch {
            print(error)
        }
        
        let productjsonStr = String(data: productjsonData, encoding: .utf8)
        let productjsonDictionary = NSDictionary(dictionary: ["viz_data" : productjsonStr!])
        
        VizuryEventLogger.logEvent("productDetails", withAttributes: productjsonDictionary as? [AnyHashable : Any])
        
        var products = [NSDictionary]()
        let productDetail1 = NSDictionary(dictionary: ["product_id" : "62112",
                                                      "quantity": "1",
                                                      "price": "50"])
        
        let productDetail2 = NSDictionary(dictionary: ["product_id" : "123123",
                                                       "quantity": "1",
                                                       "price": "50"])
        products.append(productDetail1)
        products.append(productDetail2)

        var wishListjsonData = Data()
        do {
        wishListjsonData = try JSONSerialization.data(withJSONObject: products, options: [])
        } catch {
            print(error)
        }
        let wishListjsonStr = String(data: wishListjsonData, encoding: .utf8)
        
        let wishListjsonDictionary = NSDictionary(dictionary: ["viz_data" : wishListjsonStr!])
        VizuryEventLogger.logEvent("wishlist", withAttributes: wishListjsonDictionary as? [AnyHashable : Any])
    }
}

